pkg/validator/cluster.go

```diff
@@ -312,13 +312,79 @@ func validateGeneralClusterChanges(
 	valErrors []string,
 ) []string {
 
-	if cr.Spec.AppID != prevCr.Spec.AppID {
-		appModifiedMsg := fmt.Sprintf(
-			modifiedProperty,
-			"app",
+	readyToUpgrade := shared.ClusterIsReady(prevCr)
+	upgradeInfo := prevCr.Status.UpgradeInfo
+	// If changed cluster has actual upgradeInfo (was in Updating state)
+	// we may permit the changes for rollback case only
+	// It means, the current proposed AppId should be the same as the stored in the upgradeInfo object
+	if upgradeInfo != nil && !upgradeInfo.IsRollingBack {
+		readyToUpgrade = cr.Spec.AppID == upgradeInfo.PrevApp
+	}
+
+	if !readyToUpgrade {
```

Comment:
It seems like this will fire on any change that happens if the cluster is not currently in "configured" state. I.e. we're triggering an error here before we even check to see if the user is trying to change the app. I think this should be moved down inside the `if appModified` block below.

Reply:
Hmm, shouldn't we reject any changes except app editing for rollback while cluster isn't in `configured` state?

Reply:
We discussed this out-of-band just now ... commenting here just to write it down. We do have cases where it is legal to edit the spec of a cluster that is not in configured state, and the enforcement/checking of that legality is handled elsewhere. So we shouldn't accidentally block any such legal cases here.

Reply:
@neogeographica is me, sorry. Sometimes I'm logged in with my personal account and forget to switch.

That was the discussion we had in our meeting, about cases like reverting a requested members change that can't be fulfilled.

I'll re-review the validator changes from scratch to see if there are remaining issues.